### ISSUE TITLE
Null ref: T2.MeleeWeapon.OnAttacked

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#502]
Auto bug report from 0.1.3.

Feedback IDs:
- 20200425-G644G (v0.1.3)
- 20200520-S3RP3 (v0.1.3c)

```
8:37:06, Frame 178272, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.MeleeWeapon.OnAttacked ()
T2.Weapon.OnUpdate ()
T2.CreatureBody.Update ()

GameStates: RompingThroughDungeon
GameSetupMode: Singleplayer
Players: 1
Room: 6 / 6
RoomModifiers: AxesOnly (Axe Party)
```

**Comment by ChrisNZL** [comment]
v0.1.4a

Feedback ID: 20200618-MTBJ9

```
4:21:08, Frame 37165, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.MeleeWeapon.OnAttacked ()
T2.Weapon.OnUpdate ()
T2.CreatureBody.Update ()

GameStates: RompingThroughDungeon
GameSetupMode: Singleplayer
Players: 1
DungeonRoomTransitionerState: Null
Room: 6 / 6
RoomModifiers: ExtraKnockback (击退效果加强)
YouHaveDiedState: Null
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.1.4d. Added more null checks.
